crates/wasmparser/src/validator/types.rs

```diff
@@ -504,6 +504,11 @@ impl Aliasable for AliasableResourceId {
 }
 
 impl AliasableResourceId {
+    /// Create a new instance with the specified resource and alias IDs.
+    pub fn new(id: ResourceId, alias_id: u32) -> Self {
+        Self { id, alias_id }
+    }
```

Comment:
If possible I'd prefer to avoid exporting this since the `alias_id` bits are intended to be internal implementation details which aren't exposed. 

Where this is used it looks like it's just swapping out the `ResourceId` bits, so instead of a `new` method could a method such as `fn with_resource_id(&self, id: ResourceId) -> AliasableResourceId` be added instead?